src/shapes/group.class.js

```diff
@@ -98,8 +100,19 @@
       }
 
       this.setCoords();
+      this.on('removed', this._objectCleanup);
     },
 
+    /**
+     * @private
+     */
+    _objectCleanup: function(){
+      while (this._objects.length){
+        this.remove(this._objects[0]);
+      }
+    },
```

Comment:
this and the automatic event binding are not part of the PR description.
Also this has a problem:
if you put a group inside a group, when you remove the group from the outside group, this will loose all objects.